addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -4819,6 +4923,266 @@ class SnippetsMenu extends Component {
+        this.dialog.add(ConfirmationDialog, {
+            title: _t("Install %s", snippetName),
+            body: markup(`${escape(bodyText)}\n<a href="${linkUrl}" target="_blank">${escape(linkText)}</a>`),
+            confirm: async () => {
+                try {
+                    await this.orm.call("ir.module.module", "button_immediate_install", [[moduleID]]);
+                    this.invalidateSnippetCache = true;
+                    this._onSaveRequest({
+                        data: {
+                            reloadWebClient: true,
+                        }
+                    });
+                } catch (e) {
+                    if (e instanceof RPCError) {
+                        const message = escape(_t("Could not install module %s", snippetName));
+                        this.notification.add(message, {
+                            type: "danger",
+                            sticky: true,
+                        });
+                    } else {
+                        throw e;
+                    }
+                }
+            },
+            confirmLabel: _t("Save and Install"),
+            cancel: () => {},
+        });
+    }
+    /**
+     * Deletes a custom snippet.
+     * 
+     * @private
+     * @param {Number} snippetKey
+     */
+    async _deleteCustomSnippet(snippetKey, withMutex = true) {
+        const snippet = this.snippets.get(snippetKey);
+        const message = _t("Are you sure you want to delete the block %s?", snippet.displayName);
+        this.dialog.add(ConfirmationDialog, {
+            body: message,
+            confirm: async () => {
+                await this.orm.call("ir.ui.view", "delete_snippet", [], {
+                    'view_id': snippet.id,
+                    'template_key': this.options.snippets,
+                });
+                this.invalidateSnippetCache = true;
+                this.snippets.delete(snippetKey);
+                await this._loadSnippetsTemplates(withMutex);
+            },
+            cancel: () => null,
+            confirmLabel: _t("Yes"),
+            cancelLabel: _t("No"),
+        });
+    }
+    /**
+     * Renames a custom snippet.
+     * 
+     * @private
+     * @param {Number} snippetKey
+     * @param {String} newName
+     */
+    async _renameCustomSnippet(snippetKey, newName, withMutex = true) {
+        const snippet = this.snippets.get(snippetKey);
+        if (newName !== snippet.displayName) {
+            await this.orm.call("ir.ui.view", "rename_snippet", [], {
+                'name': newName,
+                'view_id': snippet.id,
+                'template_key': this.options.snippets,
+            });
+            // Prevent the name flashing while reloading the template.
+            this.invalidateSnippetCache = newName !== snippet.displayName;
+            snippet.displayName = newName;
```

Comment:
Those lines were inside an `_execWithLoadingEffect()`, don't we want that anymore? :thinking: